### ISSUE TITLE
Bug in EarlyStopping: model checkpoint is saved even when validation loss increases

### ISSUE THREAD

**Issue by Day333** [TROUBADOUR000/TimeFilter#5]
First of all, thank you for sharing this excellent work and codebase — it is very inspiring and useful for the community.  

While reproducing the results, I noticed an issue with the `EarlyStopping` implementation. Currently, in [`EarlyStopping.__call__`]https://github.com/TROUBADOUR000/TimeFilter/blob/main/utils/tools.py, the model checkpoint is saved **unconditionally** every epoch. The logic that compares `val_loss` with the previous best (`val_loss_min`) and increments the patience counter is commented out.  

As a result:  
- The model is saved even when the validation loss **increases**, which is inconsistent with the printed message *"Validation loss decreased (...)"*.  
- This behavior may cause a mismatch between the reported performance in the paper and the reproduced results, since the checkpoint may not correspond to the actual best validation step.  

Here is an example log illustrating the issue (notice that the model is still saved when validation loss goes up from `0.702421` to `0.713448`):
```
Epoch: 4, Steps: 265 | Train Loss: 0.4495300 Vali Loss: 0.7134483 Test Loss: 0.3832419
Validation loss decreased (0.702421 --> 0.713448). Saving model ...
```
## Suggested Fix
The fix is straightforward: restore the intended early-stopping logic by checking whether the current `val_loss` is lower than the best (`val_loss_min - delta`) before saving. For example:

```python
if val_loss < self.val_loss_min - self.delta:
    self._save_checkpoint(val_loss, model, path)
    self.counter = 0
else:
    self.counter += 1
    if self.counter >= self.patience:
        self.early_stop = True
```

This bug does not fundamentally undermine the method, but it does affect reproducibility:
﻿
- Saved checkpoints may not match the actual best validation epoch.
﻿
- Reported results may differ slightly from what others obtain when following the released code.
﻿
That said, after fixing this issue in my local tests, I still find that TimeFilter remains highly competitive compared to strong baselines.

**Comment by TROUBADOUR000** [comment]
Thank you very much for your recognition and suggestions! 
It should be noted that we did not use the early stopping strategy during training. Therefore, this part is directly commented out in the source code. 
We will consider your Suggested Fix version. Thank you again!

**Comment by Day333** [comment]
> Thank you very much for your recognition and suggestions! It should be noted that we did not use the early stopping strategy during training. Therefore, this part is directly commented out in the source code. We will consider your Suggested Fix version. Thank you again!

Thank you for the clarification! That makes sense.  
Even with proper early stopping enabled, we still found that **TimeFilter** is a very strong and competitive model. Great work again!
